### PR TITLE
HIVE-27340: ThreadPool in HS2 over HTTP should respect the customized…

### PR DESCRIPTION
**Pull request by dengzhhu653**
… ThreadFactory

### Why are the changes needed?
ExecutorThreadPool in Jetty will override the customized ThreadFactory of ThreadPoolExecutor

### Does this PR introduce _any_ user-facing change?
No

### CODE REVIEW

File: service/src/java/org/apache/hive/service/cli/thrift/ThriftHttpCLIService.java [hunk 1]
```diff
@@ -94,7 +95,13 @@ protected void initServer() {
       String threadPoolName = "HiveServer2-HttpHandler-Pool";
       ThreadPoolExecutor executorService = new ThreadPoolExecutor(minWorkerThreads,
           maxWorkerThreads,workerKeepAliveTime, TimeUnit.SECONDS,
-          new SynchronousQueue<Runnable>(), new ThreadFactoryWithGarbageCleanup(threadPoolName));
+          new SynchronousQueue<Runnable>(), new ThreadFactoryWithGarbageCleanup(threadPoolName)) {
+        @Override
+        public void setThreadFactory(ThreadFactory threadFactory) {
```
Comment (aturoczy) [inline service/src/java/org/apache/hive/service/cli/thrift/ThriftHttpCLIService.java:100]: In this case would no be better to mark this method as an abstract method? Or have a parent that is abstract? To have an empty method that will be eventually overridden is bit strange to me. Maybe I don't have a context, but it seem strange :)

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java [hunk 2]
```diff
@@ -410,7 +411,12 @@ private static ThriftServer startHttpMetastore(int port, Configuration conf)
       thread.setDaemon(true);
       thread.setName("Metastore-HttpHandler-Pool: Thread-" + thread.getId());
       return thread;
-    });
+    }) {
+      @Override
+      public void setThreadFactory(ThreadFactory threadFactory) {
```
Comment (aturoczy) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java:416]: Same here

**Comment by dengzhhu653** [comment]
Some warning messages were added in the overriding method.
Hi @aturoczy, @deniskuzZ cloud you please take another look? Thanks in advance!

File: service/src/java/org/apache/hive/service/cli/thrift/ThriftHttpCLIService.java (hunk 1 above)
Comment (deniskuzZ) [reply service/src/java/org/apache/hive/service/cli/thrift/ThriftHttpCLIService.java:100]: @dengzhhu653 should it be like
````
      ThreadPoolExecutor executorService = new ThreadPoolExecutor(minWorkerThreads,
          maxWorkerThreads,workerKeepAliveTime, TimeUnit.SECONDS,
        new SynchronousQueue<>(), new ThreadFactoryWithGarbageCleanup(threadPoolName)){
            @Override
            public void setThreadFactory(ThreadFactory threadFactory) {
              if (getThreadFactory() != null) {
                LOG.warn("Ignore setting the thread factory as the pool has already provided his own: {}", getThreadFactory());
              } else {
                super.setThreadFactory(threadFactory);
              }
            }
      };
````

**Review by deniskuzZ — APPROVED** [review]
LGTM +1

**Pull request merged**